### PR TITLE
Implement progress bar shown below the player #147

### LINKED ISSUE #147: Implement progress bar shown below the player
While doing some action taking time (lumberjacking, mining) we want to show progress bar which can be interrupted for example by moving or selecting other object. 
Depends on #146

### PR DESCRIPTION
**Pull request by neu5**

### CODE REVIEW

File: client/gameObjects/Skeleton.js [hunk 1]
```diff
@@ -112,6 +116,16 @@ export default class Skeleton extends Phaser.GameObjects.Image {
       isMainPlayer
     );
 
+    this.progressBar = new ProgressBar(
```
Comment (przenz) [inline client/gameObjects/Skeleton.js]: Unless it is a progress bar for everything rename it to `ActionProgressBar`

File: server/gameObjects/Player.js [hunk 2]
```diff
@@ -47,6 +47,8 @@ class Player {
     selectedPlayerTile,
     dropSelection,
     action,
+    actionDurationTicks,
+    actionDurationMaxTicks,
```
Comment (przenz) [inline server/gameObjects/Player.js:51]: Consider turning `action` into an object with durations inside.

File: server/gameObjects/Player.js [hunk 3]
```diff
@@ -180,6 +184,12 @@ class Player {
       this.energy = 0;
       this.isDead = true;
       this.resetSelected();
+
+      if (this.action && this.actionDurationTicks !== null) {
+        this.action = null;
+        this.actionDurationTicks = null;
+        this.actionDurationMax = null;
+      }
```
Comment (przenz) [inline server/gameObjects/Player.js:192]: Turn this into a method

File: server/index.js [hunk 4]
```diff
@@ -69,6 +69,13 @@ io.on("connection", (socket) => {
             player.selectedPlayerTile = null;
           }
 
+          // action is in progress
+          if (player.action && player.actionDurationTicks !== null) {
+            player.actionDurationTicks = null;
+            player.actionDurationMax = null;
+            io.to(player.socketId).emit("action:end");
+          }
```
Comment (przenz) [inline server/index.js]: Turn this into a method, code is doubled

File: server/index.js [hunk 5]
```diff
@@ -84,6 +91,13 @@ io.on("connection", (socket) => {
         return;
       }
 
+      // action is in progress
+      if (player.action && player.actionDurationTicks !== null) {
+        player.actionDurationTicks = null;
+        player.actionDurationMax = null;
+        io.to(player.socketId).emit("action:end");
+      }
```
Comment (przenz) [inline server/index.js]: Turn this into a method, code is doubled multiple times

File: server/index.js [hunk 6]
```diff
@@ -152,29 +166,24 @@ io.on("connection", (socket) => {
...
-        io.to(player.socketId).emit("player:backpack", player.backpack);
-      }
+      player.actionDurationTicks = 0;
+      player.actionDurationMaxTicks = durationTicks;
+
+      player.energyUse(player.action);
+      io.to(player.socketId).emit(
+        "action:progress",
+        Math.floor(durationTicks * FRAME_IN_MS)
+      );
```
Comment (przenz) [inline server/index.js]: Is this correct? First `player.actionDurationMaxTicks = durationTicks;` is assigned and then `        Math.floor(durationTicks * FRAME_IN_MS)` reported as a progress?

File: server/index.js [hunk 7]
```diff
@@ -332,6 +341,22 @@ const loop = () => {
     if (player.energyRegenDelayTicks < player.energyRegenDelayMaxTicks) {
       player.energyRegenDelayTicks += 1;
     }
+
+    if (player.action && player.actionDurationTicks !== null) {
```
Comment (przenz) [inline server/index.js]: This `if` occurs multiple times.

File: server/index.js [hunk 8]
```diff
@@ -332,6 +341,22 @@ const loop = () => {
...
+      if (player.actionDurationTicks < player.actionDurationMaxTicks) {
+        player.actionDurationTicks += 1;
+      } else {
+        player.actionDurationTicks = null;
+        player.actionDurationMax = null;
```
Comment (przenz) [inline server/index.js]: Again action reset, use method

**Review by przenz — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 6 above)
Comment (neu5) [reply server/index.js]: yes, it's correct
duractionTicks is in ticks... it's not miliseconds or seconds, it is just a number.
By the multiplication `durationTicks * FRAME_IN_MS` we get miliseconds that are required by tweens later on the client side.
Our current FRAME_IN_MS is 1000/30 = 33,(3).
Eg. 150 * 33,3 = 4995 ms, which is 5s, which is good :)

Comment (przenz) [reply server/index.js]: Actually I was asking about the progress itself, but now I can see that this is actually not reporting progress but starting counter on client side instead:
```
  game.socket.on("action:progress", (duration) => {
    game.mainPlayer.actionStart(duration);
  });
```
Please rename it to `action:start` or `action:progressStart`

Comment (neu5) [reply server/index.js]: Yes, it's just the value for telling the progress bar how long the animation should take.
I renamed it :)

**Review by przenz — APPROVED** [review]

**Pull request merged**